lib/Twig/Environment.php

```diff
@@ -767,6 +761,11 @@ public function addExtension(Twig_ExtensionInterface $extension)
             throw new LogicException(sprintf('Unable to register extension "%s" as extensions have already been initialized.', $extension->getName()));
         }
 
+        $r = new ReflectionObject($extension);
+        if (($extensionTime = filemtime($r->getFileName())) > $this->lastModifiedExtension) {
+            $this->lastModifiedExtension = $extensionTime;
+        }
```

Comment:
this add a check even when auto_reload is false. Are you sure this has less perf impact ? To me, it has more impact.

Resetting the value should not have been an issue anyway: it is forbidden to add more extension after you started rendering templates